src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -33,162 +43,465 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
+            }
+
+            void ReportDiagnosticsForSymbol(ISymbol symbol, string symbolName, Action<Diagnostic> reportDiagnostic, bool checkForUnmeaningful = true)
+            {
+                foreach (var misspelledWord in GetMisspelledWords(symbolName))
+                {
+                    reportDiagnostic(GetMisspelledWordDiagnostic(symbol, misspelledWord));
+                }
+
+                if (checkForUnmeaningful && symbolName.Length == 1)
+                {
+                    reportDiagnostic(GetUnmeaningfulIdentifierDiagnostic(symbol, symbolName));
+                }
+            }
+
+            IEnumerable<string> GetMisspelledWords(string symbolName)
+            {
+                var parser = new WordParser(symbolName, WordParserOptions.SplitCompoundWords);
+                if (parser.PeekWord() != null)
+                {
+                    var word = parser.NextWord();
+                    if (word == null)
+                    {
+                        yield break;
+                    }
+
+                    do
+                    {
+                        if (IsWordAcronym(word) || IsWordNumeric(word) || IsWordSpelledCorrectly(word))
+                        {
+                            continue;
+                        }
+
+                        yield return word;
+                    }
+                    while ((word = parser.NextWord()) != null);
+                }
+            }
+
+            static bool IsWordAcronym(string word) => word.All(char.IsUpper);
+
+            static bool IsWordNumeric(string word) => char.IsDigit(word[0]);
+
+            bool IsWordSpelledCorrectly(string word)
+                => !projectDictionary.UnrecognizedWords.Contains(word) && projectDictionary.RecognizedWords.Contains(word);
+        }
+
+        private static CodeAnalysisDictionary GetMainDictionary()
+        {
+            // The "main" dictionary, Dictionary.dic, was created in WSL Ubuntu with the following commands:
+            //
+            // Install dependencies:
+            // > sudo apt install hunspell-tools hunspell-en-us
+            // 
+            // Create dictionary:
+            // > unmunch /usr/share/hunspell/en_US.dic /usr/share/hunspell/en_US.aff > Dictionary.dic
+            //
+            // Tweak:
+            // Added the words: 'namespace'
+            var text = SourceText.From(TextAnalyzersResources.Dictionary);
+            return ParseDicDictionary(text);
+        }
+
+        private static CodeAnalysisDictionary ParseXmlDictionary(SourceText text)
+            => text.Parse(CodeAnalysisDictionary.CreateFromXml);
+
+        private static CodeAnalysisDictionary ParseDicDictionary(SourceText text)
+            => text.Parse(CodeAnalysisDictionary.CreateFromDic);
+
+        private static string RemovePrefixIfPresent(string prefix, string name)
+            => name.StartsWith(prefix, StringComparison.Ordinal) ? name[1..] : name;
```

Comment:
All use of this method only go for a `char` prefix

```suggestion
        private static string RemovePrefixIfPresent(char prefix, string name)
            => name.Length > 0 && name[0] == prefix ? name[1..] : name;
```

Reply:
Done